### PR TITLE
TK3.10: get-in-config in migration tests, better resource management

### PR DESCRIPTION
**Pull request by frenchy64**
**Epic** https://github.com/threatgrid/iroh/issues/3986

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: TK3.10: get-in-config in migration tests, better resource management
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/task/migration/migrate_es_stores_test.clj [hunk 1]
```diff
@@ -60,23 +60,24 @@
...
+(defn es-conn [get-in-config]
+  (connect (:default (es-props get-in-config))))
 
-;; This a is a `defn` to prevent side-effects at compile time
-(defn migration-index []
-  (p/get-in-global-properties [:ctia :migration :store :es :migration :indexname]))
+(defn migration-index [get-in-config]
+  (get-in-config [:ctia :migration :store :es :migration :indexname]))
 
 (defn fixture-clean-migration [t]
```
Comment (frenchy64) [inline test/ctia/task/migration/migrate_es_stores_test.clj:72]: FWIW an almost identical fixture lives in the other file in this PR, but refers to a different `es-conn`.

File: test/ctia/task/migration/store_test.clj [hunk 2]
```diff
@@ -248,15 +247,24 @@
...
-(def es-conn (connect (:default es-props)))
-(def migration-index (get-in es-props [:migration :indexname]))
+(defn es-props [get-in-config]
+  (get-in-config [:ctia :store :es]))
+
+(defn es-conn [get-in-config]
+  (connect (:default (es-props get-in-config))))
+
+(defn migration-index [get-in-config]
+  (get-in (es-props get-in-config) [:migration :indexname]))
```
Comment (frenchy64) [inline test/ctia/task/migration/store_test.clj:257]: I noticed this `migration-index` function differs slightly from the one in migrate-es-stores-test:

```clojure
(defn migration-index [get-in-config]
  (get-in-config [:ctia :migration :store :es :migration :indexname]))
```

Intentional?

**Review by vemv — APPROVED** [review]
lgtm!

**Review by gbuisson — APPROVED** [review]
LGTM!

**Comment by frenchy64** [comment]
Thanks for the reviews!

**Pull request merged**